beacon_node/network/src/service.rs

```diff
@@ -733,6 +737,17 @@ impl<T: BeaconChainTypes> NetworkService<T> {
                     }
                 }
 
+                // TODO(das): This is added here for the purpose of testing, *without* having to
+                // activate Electra. This should happen as part of the Electra upgrade and we should
+                // move the subscription logic once it's ready to rebase PeerDAS on Electra, or if
+                // we decide to activate via the soft fork route:
+                // https://github.com/sigp/lighthouse/pull/5899
+                let peer_das_scheduled = self.fork_context.spec.eip7594_fork_epoch
```

Comment:
In `ChainSpec` we interpret a `None` on a fork epoch as equivalent to `FAR_FUTURE_EPOCH` (fork unset). So this could incorrectly return true

Reply:
Thanks, I've added a `spec.is_peer_das_scheduled` function that checks for both `None` and `FAR_FUTURE_EPOCH` values. 
https://github.com/sigp/lighthouse/commit/94f1d3dd93622b358e52b1ee79febbed67292c84